### PR TITLE
Cursed parrot rewrite. [MDB IGNORE] [IDB IGNORE]

### PR DESCRIPTION
**Pull request by MistakeNot4892**
## Description of changes

Depends on #3008

Rewrote parrots to implement inventory procs and use datum AI. This might be a good use case for implementing FSM AI in the future since the states are now fairly clearly defined.

## Why and what will this PR improve
Parrot code bad.

## Authorship
Myself.

## Changelog
:cl:
tweak: Parrots now use intents for determining if they grab, attack or perch, rather than verbs.
/:cl:

### CODE REVIEW

**Comment by MistakeNot4892** [comment]
Outstanding issues:
- Perching behavior is not working for the AI, but works alright for player controlled.
- Moving as a player controlled parrot isn't clearing resting status like it should.

File: code/modules/mob/living/simple_animal/parrot/_parrot.dm
Comment (MistakeNot4892) [inline code/modules/mob/living/simple_animal/parrot/_parrot.dm]: Should be `ispath`.

File: code/modules/mob/living/simple_animal/parrot/_parrot.dm
Comment (MistakeNot4892) [inline code/modules/mob/living/simple_animal/parrot/_parrot.dm]: `isobj`

**Comment by MistakeNot4892** [comment]
I'm calling this done, I'll fix whatever's left after the telecomms PR goes in.

File: code/modules/mob/living/simple_animal/simple_animal.dm
Comment (MistakeNot4892) [inline code/modules/mob/living/simple_animal/simple_animal.dm]: This is an error.

File: code/modules/mob/living/simple_animal/parrot/_parrot.dm
Comment (MistakeNot4892) [inline code/modules/mob/living/simple_animal/parrot/_parrot.dm:187]: This will possibly be out of date by the time this PR goes in.

File: code/datums/ai/ai.dm
Comment (MistakeNot4892) [inline code/datums/ai/ai.dm]: This got cherrypicked I think, needs to be checked.

File: code/modules/mob/living/simple_animal/parrot/_parrot.dm
Comment (out-of-phaze) [inline code/modules/mob/living/simple_animal/parrot/_parrot.dm:67]: Nonzero chance that not syncing `animate_movement` here might cause some visual issues; I had to deal with that when working on vehicles at one point.

File: code/modules/mob/living/simple_animal/parrot/_parrot.dm
Comment (out-of-phaze) [inline code/modules/mob/living/simple_animal/parrot/_parrot.dm]: This will stop parrots from perching on objects they're on the same tile as, which I don't think is intended.

**Review by out-of-phaze — CHANGES_REQUESTED** [review]
Only one small concern about perching. I don't really like the defines used in the parrot AI code but it also kind of went over my head anyway, so I just skimmed it.

File: code/modules/mob/living/simple_animal/parrot/_parrot.dm
Comment (MistakeNot4892) [reply code/modules/mob/living/simple_animal/parrot/_parrot.dm:67]: Not super clear on where I would reset this for the parrot.

File: code/modules/mob/living/simple_animal/parrot/_parrot.dm
Comment (out-of-phaze) [inline code/modules/mob/living/simple_animal/parrot/_parrot.dm:104]: Should probably reset `animate_movement` to its initial value here?

**Comment by MistakeNot4892** [comment]
Going to rebase this on #3081 when it's ready to go.

**Comment by out-of-phaze** [comment]
What's the status on this one? Blocked by/waiting for any other PRs?

**Comment by MistakeNot4892** [comment]
Probably good to go when I rebase it.

**Comment by MistakeNot4892** [comment]
I managed to make this depend on mob inventory slots more broadly so will revisit when that stuff is done. -_-

**Pull request closed without merging**
